### ISSUE TITLE
Glide64: Mem. Leak (?) when using Software Depth Buffer Calc.

### ISSUE THREAD

**Issue by BrendonIrwan** [project64/project64#1486]
First off, I'm unsure whether you have any connections to Glide64 devs. If you do, then please show me the way and I will forward this report to them. Anyways,

**Problem**
I recently turned the aforementioned option on for some reason. After playing around Conker BFD for some time, Windows began complaining about the lack of available memory. No other programs were actively working at that time so I turned to PJ64.

**Diagnosis**
I restarted the emu, this time observing how memory is consumed. It began when entering the main game (i.e. after loading a file), so I looked into the plugin. For each non-default setting I did, I turn it off, run the game (from a fresh instance), observe, and turn it back on if nothing changes. I came into the conclusion that the Software Depth Buffer Calculation was causing the issue.

What's weird is that according to Task Manager, PJ64's mem usage is ok (~200 MB when running). So it may have no relevance to you, but I'm reporting it here anyways.

**Solution**
Turn it off, for now.

**Environment**
Project64 2.3.2.202
Intel HD Graphics 620

Part of my Project64.cfg (with symptom turned off):
```
[Plugin]
Audio Dll Ver=Jabo's DirectSound 1.7.0.7
Controller Dll Ver=N-Rage For PJ64: 2.5.2.202
Graphics Dll=GFX\PJ64Glide64.dll
Graphics Dll Ver=Glide64 For PJ64: 2.1.2.202
RSP Dll Ver=RSP Plugin 1.7.3.202

[Glide64]
clock=1
clock_24_hr=1
ghq_cache_save=0
ghq_cache_size=128
ghq_enht=3
lodmode=2
ssformat=0
texenh_options=1
```

**Comment by theboy181** [comment]
how big is your PJ64 directory now?

**Comment by oddMLan** [comment]
Sounds like the VRAM is getting filled up.

**Comment by BrendonIrwan** [comment]
How big is my PJ64 dir? It's a fresh install onto a 128 MB ramdisk, so pretty small.

VRAM getting filled up? Probably.

**Comment by theboy181** [comment]
I have had an error log build up (not sure if it was Glide64) that was over 2GB. just saying.

**Comment by BrendonIrwan** [comment]
Before anyone may ask, both my RAM and pagefile.sys are 4 GB.

**Comment by oddMLan** [comment]
Since you got an IGP, and if I'm right this is a VRAM matter, shared memory should be around 64MB-128MB, which is probably a tad low. Although a bigger VRAM would only mask the problem, if the depth copies aren't getting cleared properly, it would just take longer to experience the same issue.

**Comment by BrendonIrwan** [comment]
From what I know--I may be wrong--shared VRAM is limited up to 128 MB (at least on my HD Graphics 4400), whatever the case is, at least the problem doesn't break gameplay do they?

I'll do some testing with GLideN64 later--if I can get it running on PJ64. I never use any plugins outside what is bundled.

For clarity sake, I'll also try to fire up my DGPU (barely use it anyways so it needs a warm up) with both Glide64 and GLideN64.

**Comment by BrendonIrwan** [comment]
So I did a test on my older laptop with Intel HD Graphics 4400 and NVIDIA GeForce 840m (Iame I know). Results:

Note: SDBR is short for Software depth buffer rendering

**Intel HD Graphics 4400**
SDBR ON - Rapid RAM consumption. Game freezes after a minute. After that UI still responsive but breaks after some clicks
SBDR OFF - Normal memory consumption

**NVIDIA GeForce 840m**
SDBR ON - Rapid pagefile.sys consumption (probs the DGPU has its own VRAM?)
SDBR OFF - Normal memory consumption

Then one of you ask me to try it on GlideN64. Under default settings, it works okay (on Intel).

**Additional Notes**
I monitor system usage via this Rainmeter widget I made some time ago. It looks like this:
![image](https://user-images.githubusercontent.com/13151306/44824359-a4cbf500-abf4-11e8-9b29-bf3cd5a34610.png)
I confirmed that the information matches to what is shown in Task Manager. I cannot monitor VRAM usage unless I update to Fall Creators. The code for this stuff is on my GitHub (edenrm).

I also observed that the memory consumption happens when during gameplay, cutscenes don't affect that much.

BTW, loading a ROM while the emu is initializing for another ROM will crash it.
